### PR TITLE
[NTUSER] Add locking at IntSendFocusMessage

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Fix access violation on `IntFocusSetInputContext`.
JIRA issue: [CORE-18163](https://jira.reactos.org/browse/CORE-18163)

## Proposed changes

- Add locking on `pWndPrev`.

### CODE REVIEW

File: win32ss/user/ntuser/focus.c [hunk 1]
```diff
@@ -614,17 +614,24 @@ co_IntSendActivateMessages(PWND WindowPrev, PWND Window, BOOL MouseActivate, BOO
...
    {
       ThreadQueue->QF_flags |= QF_FOCUSNULLSINCEACTIVE;
    }
 
-   pWndPrev = ThreadQueue->spwndFocus;
+   pRealWndPrev = pWndPrev = ThreadQueue->spwndFocus;
+   if (pRealWndPrev)
+   {
+      UserRefObjectCo(pRealWndPrev, &Ref);
+      if (!ValidateHwndNoErr(UserHMGetHandle(pRealWndPrev)))
```
Comment (HBelusca) [inline win32ss/user/ntuser/focus.c:632]: Why is that handle stuff needed? Shouldn't it be that from within running win32k code, all these stored non-NULL PWNDs are automatically valid?

Comment (katahiromz) [reply win32ss/user/ntuser/focus.c:632]: `spwndFocus` might be invalid.

Comment (HBelusca) [reply win32ss/user/ntuser/focus.c:632]: In what circumstances? I would expect it to be either NULL or a pointer to a valid PWND ?

Comment (katahiromz) [reply win32ss/user/ntuser/focus.c:632]: Ideally `spwndFocus` should be assignment-locked but ReactOS has different impl.

Comment (katahiromz) [reply win32ss/user/ntuser/focus.c:632]: This function is called from `co_UserSetFocus`.

Comment (katahiromz) [reply win32ss/user/ntuser/focus.c:632]: The below code sending {{WM_KILLFOCUS}}.

Comment (katahiromz) [reply win32ss/user/ntuser/focus.c]: This code was useless. Deleted.

Comment (katahiromz) [reply win32ss/user/ntuser/focus.c]: This code was useless. Deleted.

**Review by HBelusca — COMMENTED** [review]
If this solves the reported problem, without breaking anything else, then LGTM :)

File: win32ss/user/ntuser/focus.c (hunk 1 above)
Comment (learn-more) [reply win32ss/user/ntuser/focus.c]: `co_` functions do a callback to usermode code, so the entire world can change during those ;)

**Pull request merged**